### DISCUSSION TITLE
Adding ssh key based authentication as a way to connect

### DISCUSSION THREAD

**Discussion by Lolozendev** [community-scripts/ProxmoxVE#860] [Ideas]
Why not add a way to choose between password and keybased authentication. public key ssh authentication , is a much more secure and safe way to authenticate to a server, it's already supported by all operating systems and I think it's much more manageable than creating a random password for each new container or vm. What do you think?

**Comment by MickLesk** [comment]
Do an PR and we can discuss

**Comment by Mellowlynx** [comment]
Some use it, most don't. 

I agree that it would be nice to have (especially if password login for the root user will also be disabled) and have way better security. 

But I'm afraid that most users do not have the know-how to support this out of the box.

Myself I use Ansible to deploy my key-sync script and motd, etc. to new nodes and hosts, maybe something to take a look at.

And yes, make a PR request so we can discuss it further.

**Comment by Lolozendev** [comment]
I'm trying to create the PR but I must be a collaborator (never opened one maybe I'm not doing properly sorry)

**Reply by dsiebel** [reply]
> I'm trying to create the PR but I must be a collaborator (never opened one maybe I'm not doing properly sorry)

You need to fork the repository and create the PR from there.

**Comment by dsiebel** [comment]
I took a first stab at this today:

~https://github.com/community-scripts/ProxmoxVE/pull/1052~
https://github.com/community-scripts/ProxmoxVE/pull/1060

**Reply by Lolozendev** [reply]
amazing work, I haven't had much time to work on this lately sorry, but your solution is very nice !

**Comment by dgm** [comment]
I for one was just struggling with this a few days ago - I'm a newcomer to proxmox but I've been administering linux systems using public key ssh for well over a decade.

I'm also struggling with file manipulation inside the containers,  but having my ssh access fixed would probably help tremendously.

**Comment by Koliera** [comment]
I've mitigated this with using tailscale for accessing the LXC - allows ssh & ftp as needed (based on the options selected).

**Comment by MickLesk** [comment]
merged with #1456
